docs/kb/semgrep-ci/azure-using-templates-with-semgrep.md

```diff
@@ -0,0 +1,75 @@
+---
+tags:
+  - Azure
+  - Templates
+description: Using templates in Azure pipelines with Semgrep commands.
+---
+# Using templates in Azure pipelines with Semgrep
+
+## Motivation
+
+Complex code can take more work to maintain and modify. So, encapsulating some functionalities in functions, methods, or external files is a good programming practice.
+That is the primary goal of Azure templates: extract some functionalities (snippets) defined in a big yaml file to an external file (template). With that, we are improving pipelines' readability and maintainability.
+
+## Defining Semgrep snippets in a template file
+
+Adding semgrep commands in a yaml template is a simple task. In the example, we will create two templates, one for Semgrep full scan and one for Semgrep pull request scans.
+
+Example for Semgrep full scan in a template yaml file:
+``````
+steps:
+- script: |
+    echo "Semgrep full scan"
+    python -m pip install --upgrade pip
+    pip install semgrep
+    semgrep ci
+``````
+
+Example for Semgrep pull request scan in a template yaml file:
```

Comment:
I can also take care of indenting this and adding it to the ordered list.

Reply:
Yes, please.